### PR TITLE
serialise/deserialise added for flow_nums consistency

### PR DESCRIPTION
**Pull request by datamel**
These changes close https://github.com/cylc/cylc-flow/issues/4672

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] Already covered by existing tests.
- [x] No change log entry required (why? invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Review by hjoliver — APPROVED** [review]
Looks good :+1:  Could you just add a comment in the new function to say why we're wrapping `json.dumps` - i.e., to get a sorted result.

File: tests/unit/test_util.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from cylc.flow.util import deserialise
+
+
+def test_deserialise():
+    actual = deserialise('["2", "3"]')
+    expected = {'2', '3'}
+    assert actual == expected
```
Comment (oliver-sanders) [inline tests/unit/test_util.py:23]: The doctest is already running this example?

**Review by oliver-sanders — APPROVED** [review]
Looks good.

File: tests/unit/test_util.py (hunk 1 above)
Comment (datamel) [reply tests/unit/test_util.py:23]: Woops that is a mistake, thanks, I'll remove the doctest. Can't test set comparison in doctest as it does a literal check.

Comment (oliver-sanders) [reply tests/unit/test_util.py:23]: Aah.

**Comment by oliver-sanders** [comment]
(letting tests re-run)

**Pull request merged**